File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int j=0;
        int mx=0;
        for(int i=0;i<s.length();i++){
            String t="";
            j=i;
            while(j<s.length() && (!t.contains(String.valueOf(s.charAt(j))))){
                t+=s.charAt(j);
                j++;
            }
            mx=Math.max(mx,t.length());
        }
        return mx;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> m;
        int r=0;
        int l=0;
        int mx=0;
        while(r<s.size()){
            m[s[r]]++;
            while(l<r && m[s[r]]>1){
                m[s[l]]--;
                if(m[s[l]]==0) m.erase(s[l]);
                l++;
            }
            mx=max(mx,r-l+1);
            r++;
        }
        return mx;
    }
};